lib/htmlbars-plugin/index.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+var utils = require('./utils');
+
+function ClassTransformPlugin() {
+  this.syntax = null;
+  this.builders = null;
+}
+
+ClassTransformPlugin.prototype.constructor = ClassTransformPlugin;
+
+ClassTransformPlugin.prototype.transform = function(ast) {
+  if (!this.builders) {
+    this.builders = this.syntax.builders;
+  }
+
+  var plugin = this;
+  var walker = new this.syntax.Walker();
+
+  walker.visit(ast, function(node) {
+    switch(node.type) {
+      case 'ElementNode':
+        plugin.transformElementNode(node);
+        break;
+      case 'MustacheStatement':
+      case 'BlockStatement':
+        plugin.transformStatement(node);
+        break;
+    }
+  });
+
+  return ast;
+};
+
+ClassTransformPlugin.prototype.transformStatement = function(node) {
+  var localClassPair = utils.getPair(node, 'local-class');
+  if (!localClassPair) { return; }
+
+  utils.removePair(node, localClassPair);
+
+  var classPair = utils.getPair(node, 'class');
+  var params = [];
+  var concatSexpr;
+
+  if (classPair) {
+    params.push(classPair.value);
+    utils.removePair(node, classPair);
+  }
+
+  utils.pushAll(params, this.localToPath(localClassPair.value.value));
+  this.divide(params);
+  concatSexpr = this.builders.sexpr('concat', params);
+  node.hash.pairs.push(this.builders.pair('class', concatSexpr));
+};
+
+ClassTransformPlugin.prototype.transformElementNode = function(node) {
+  var localClassAttr = utils.getAttr(node, 'local-class');
+  if (!localClassAttr) { return; }
+
+  utils.removeAttr(node, localClassAttr);
+
+  var classAttr = utils.getAttr(node, 'class');
+  var parts = [];
+  var classAttrValue
+
+  if (classAttr) {
+    utils.removeAttr(node, classAttr);
+    classAttrValue = classAttr.value;
+
+    // Unwrap the original class attribute value and transform
+    // the attr value into parts that make up the ConcatStatement
+    // that is returned from this method
+    if (classAttrValue.type === 'ConcatStatement') {
+      parts.push(this.builders.sexpr('concat', classAttrValue.parts));
+    } else if (classAttrValue.type === 'TextNode') {
+      parts.push(this.builders.string(classAttrValue.chars));
+    } else if (classAttrValue.type === 'MustacheStatement') {
+      if (classAttrValue.params.length) {
+        parts.push(classAttrValue);
+      } else {
+        parts.push(this.builders.path(classAttrValue.path.original));
```

Comment:
I'll happily supported it once I figure out a way to validate it without writing a custom helper.  If an issue gets opened, I'll take it up I'm just unsure, like you said about, what it would look like when used with an ElementNode.